Aptoide/aptoide-client-v8
# [APP-2865] Navigation by app id

**What does this PR do?**

   - Changes AppViewScreen navigation to use app id instead of package name when possible.
   - Removes the flag that controls store name usage on app meta requests. The store name is now only appended when the request is done with anything other than app id.
   - Makes the installer notifications use app id instead of package name when creating the deeplink to navigate to the app.

**Database changed?**

   Yes - changed the AppDetails database in order to include the app id. It will be the app id of the version for which an installation was started.

**Where should the reviewer start?**


**How should this be manually tested?**

  **Every flow that goes into the app view should be tested. That includes navigating from:**

1. Home Screen
2. See all screens
3. Ahab - atm only notifications
4. APKFY

  Flow on how to test this or QA Tickets related to this use-case: [APP-2865](https://aptoide.atlassian.net/browse/APP-2865)


**What are the relevant tickets?**

  Tickets related to this pull-request: [APP-2865](https://aptoide.atlassian.net/browse/APP-2865)

**Code Review Checklist**


[APP-2865]: https://aptoide.atlassian.net/browse/APP-2865?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[APP-2865]: https://aptoide.atlassian.net/browse/APP-2865?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
